ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -1133,7 +1133,20 @@ private String processTable(QB qb, ASTNode tabref) throws SemanticException {
 
     if (asOfTimeIndex != -1 || asOfVersionIndex != -1) {
       String asOfVersion = asOfVersionIndex == -1 ? null : tabref.getChild(asOfVersionIndex).getChild(0).getText();
-      String asOfTime = asOfTimeIndex == -1 ? null : tabref.getChild(asOfTimeIndex).getChild(0).getText();
+      String asOfTime = null;
+      
+      if (asOfTimeIndex != -1) {
+        ASTNode expr = (ASTNode) tabref.getChild(asOfTimeIndex).getChild(0);
+        if (expr.getChildCount() > 0) {
+          ExprNodeDesc desc = genExprNodeDesc(expr, null, false, true);
+          if (desc instanceof ExprNodeConstantDesc) {
+            ExprNodeConstantDesc c = (ExprNodeConstantDesc) desc;
+            asOfTime = String.valueOf(c.getValue());
+          }
```

Comment:
Should an exception be thrown if the expression can not be evaluated at compile time?

It might be worth adding some negative test cases to cover this code path.

Reply:
👍